src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -77,50 +84,136 @@ class CrossDeviceLink extends Component {
   }
 
   render = () =>
-    this.props.roomId ? <CrossDeviceLinkUI roomId={this.props.roomId} /> : <Spinner />
+    this.props.roomId ? <CrossDeviceLinkUI {...this.props}/> : <Spinner />
 }
 
 class CrossDeviceLinkUI extends Component {
   constructor(props) {
     super(props)
-    this.state = {copySuccess: false}
+    this.state = {
+      copySuccess: false,
+      sending: false,
+      mobileNumber: null,
+      error: {}
+    }
   }
 
+  linkId = `${process.env.BASE_36_VERSION}${this.props.roomId}`
+
+  linkCopiedTimeoutId = null
+
   copyToClipboard = (e) => {
+    e.preventDefault()
     this.textArea.select()
     document.execCommand('copy')
     e.target.focus()
     this.setState({copySuccess: true})
+    this.clearLinkCopiedTimeout()
+    this.linkCopiedTimeoutId = setTimeout(() => {
+      this.setState({copySuccess: false})
+    }, 5000)
```

Comment:
Why does copying something require a timeout?

Is 5 seconds quite long given there's no network dependency?

Reply:
Because as discussed with @danielspagnolo, we want to support multiple copy attempts. If they copy once, then copy something else, we want to give them the possibility to copy the link again.

Reply:
oh ok i understand now - nifty feature